pkg/apis/common/v1/association.go

```diff
@@ -106,6 +109,10 @@ const (
 	// should use `SingletonAssociationID` as their `AssociationID`. On the contrary, Agent can have unbounded number
 	// of Associations so Agent-ES Associations should _not_ use `SingletonAssociationID`.
 	SingletonAssociationID = ""
+
+	// NoAuthRequiredValue is the value set for AuthSecretName if no authentication
+	// credentials are necessary for that association.
+	NoAuthRequiredValue = "-"
```

Comment:
I'm assuming here that no secret will ever be named "-". Since this is a value we set with our own secret naming scheme it should be OK.
It's somewhat nice because it doesn't break the existing schema and is quite easy to manipulate in the code.
However it also feels a bit "implicit", we could also modify the assoc conf schema to make things more explicit. WDYT?